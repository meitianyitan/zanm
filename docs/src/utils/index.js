function iframeReady(iframe, callback) {
  const doc = iframe.contentDocument || iframe.contentWindow.document;
  const interval = () => {
    if (iframe.contentWindow.changePath) {
      callback();
    } else {
      setTimeout(() => {
        interval();
      }, 50);
    }
  };

  if (doc.readyState === 'complete') {
    interval();
  } else {
    iframe.onload = interval;
  }
}

const ua = navigator.userAgent.toLowerCase();
const isMobile = /ios|iphone|ipod|ipad|android/.test(ua);

export {
  isMobile,
  iframeReady
};
