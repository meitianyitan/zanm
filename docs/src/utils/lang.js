import Locale from '../../../packages/locale';
import zhCN from '../../../packages/locale/lang/zh-CN';
import enUS from '../../../packages/locale/lang/en-US';

const langMap = {
  'en-US': {
    title: 'Zanm - A Mobile UI for Vue.js 2.0',
    messages: enUS
  },
  'zh_CN': {
    title: 'Zanm - 一套基于 Vue.js 的移动端组件库',
    messages: zhCN
  }
};
let currentLang = '';

setLang(getDefaultLang());

function getDefaultLang() {
  const langs = Object.keys(langMap);
  const hash = location.hash;

  for (let i = 0; i < langs.length; i++) {
    if (~hash.indexOf(langs[i])) {
      return langs[i];
    }
  }

  const userLang = localStorage.getItem('ZANM_LANGUAGE') || navigator.language || 'en_US';
  return ~userLang.indexOf('zh-') ? 'zh_CN' : 'en_US';
}

export function setLang(lang) {
  if (currentLang === lang) {
    return;
  }

  currentLang = lang;
  if (window.localStorage) {
    localStorage.setItem('ZANM_LANGUAGE', lang);
  }
  Locale.use(lang, langMap[lang].messages);
  document.title = langMap[lang].title;
}
