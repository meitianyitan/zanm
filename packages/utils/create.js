/**
 * Create a component with common options
 */
import createBasic from './create-basic';
import Loading from '../loading';

export default function(sfc) {
  sfc.components = Object.assign(sfc.components || {}, {
    Loading
  });
  return createBasic(sfc);
};
