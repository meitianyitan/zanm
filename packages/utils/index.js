function isDef(value) {
  return value !== undefined && value !== null;
}

function isObj(x) {
  const type = typeof x;
  return x !== null && (type === 'object' || type === 'function');
}

function get(object, path) {
  const keys = path.split('.');
  let result = object;

  keys.forEach(key => {
    result = isDef(result[key]) ? result[key] : '';
  });

  return result;
}

const camelizeRE = /-(\w)/g;
function camelize(str) {
  return str.replace(camelizeRE, (_, c) => c.toUpperCase());
}

export {
  isDef,
  isObj,
  get,
  camelize
};
