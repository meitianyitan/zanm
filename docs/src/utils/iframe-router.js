/**
 * 同步父窗口和 iframe 的 vue-router 状态
 */
import { setLang } from './lang';
import { iframeReady, isMobile } from './';

window.syncPath = function(dir) {
  const router = window.vueRouter;
  const isInIframe = window !== window.top;
  const currentDir = router.history.current.path;
  const pathParts = currentDir.split('/');
  let lang = pathParts[0];
  if (currentDir[0] === '/') {
    lang = pathParts[1];
  }

  if (!isInIframe && !isMobile) {
    const iframe = document.querySelector('iframe');
    if (iframe) {
      iframeReady(iframe, () => {
        iframe.contentWindow.changePath(lang, currentDir);
      });
    }
    setLang(lang);
  } else if (isInIframe) {
    window.top.changePath(lang, currentDir);
  }
};

window.changePath = function(lang, path = '') {
  setLang(lang);
  window.vueRouter.replace(path);
};
