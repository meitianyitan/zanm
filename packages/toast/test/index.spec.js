import Toast from '../';
import { transitionStub, later } from '../../../test/utils';

transitionStub();

test('create a forbidClick toast', async() => {
  const toast = Toast({
    forbidClick: true
  });

  expect(toast.$el.outerHTML).toMatchSnapshot();

  await later();
  expect(document.body.classList.contains('zvm-toast--unclickable')).toBeTruthy();
  toast.forbidClick = false;

  await later();
  expect(document.body.classList.contains('zvm-toast--unclickable')).toBeFalsy();
});

it('toast disappeared after duration', async() => {
  const toast = Toast({
    duration: 10
  });

  await later(50);
  expect(toast.$el.style.display).toEqual('none');
});

test('clear toast', () => {
  const toast1 = Toast();
  expect(toast1.value).toBeTruthy();
  Toast.clear();
  expect(toast1.value).toBeFalsy();

  Toast.allowMultiple();
  const toast2 = Toast('2');
  const toast3 = Toast('3');
  Toast.clear(true);
  expect(toast2.value).toBeFalsy();
  expect(toast3.value).toBeFalsy();
  Toast.allowMultiple(false);
});

test('multiple toast', () => {
  Toast.allowMultiple();
  Toast.clear(true);
  const toast1 = Toast.success('1');
  const toast2 = Toast.success('2');
  Toast.clear();
  expect(toast1.value).toBeFalsy();
  expect(toast2.value).toBeTruthy();
  Toast.clear();
  Toast.clear();
  expect(toast2.value).toBeFalsy();
  Toast.allowMultiple(false);
});

test('set default options', () => {
  Toast.setDefaultOptions({ duration: 1000 });
  expect(Toast().duration).toEqual(1000);
  Toast.resetDefaultOptions();
  expect(Toast().duration).toEqual(3000);
});

test('toast duration 0', () => {
  Toast.allowMultiple();
  const toast = Toast({
    message: 'toast',
    duration: 0
  });
  expect(toast.timer).toBeFalsy();
  Toast.allowMultiple(false);
});
