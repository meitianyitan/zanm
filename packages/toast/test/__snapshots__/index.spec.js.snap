// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a forbidClick toast 1`] = `
<div class="zvm-toast zvm-toast--text zvm-toast--middle" style="display: none;" name="zvm-fade">
  <div></div>
  <!---->
  <!---->
</div>
`;
